compiler/rustc_driver/src/lib.rs

```diff
@@ -1288,8 +1288,6 @@ pub fn init_env_logger(env: &str) {
         .with_indent_lines(true)
         .with_ansi(true)
         .with_targets(true)
-        .with_thread_ids(true)
```

Comment:
I did turn these on on purpose, because in a parallel rustc we'll be unable to pick apart the tracing messages otherwise. One alternative would be to only turn them on if there's a `RUSTC_LOG_THREADS` env var set. So by default we would not emit these for now. What do you think?

Reply:
Is there a way to tell if the parallel compiler is enabled? Maybe with a `cfg`? It would be nice for this to be enabled/disabled automatically.

Reply:
it's a command line flag, which we don't have access to at this stage. I think we can cross that bridge when parallel compiling and debuggin parallel compilers becomes more common.

Reply:
@oli-obk are you sure? I see a lot of `cfg(parallel_compiler)` around the code base - do you have to _both_ build the parallel compiler _and_ opt into it with a command line flag?

Reply:
There's two bits of parallel compiler: query-level support in compiled code (e.g. using a mutex), and separately how many threads we actually end up using (-Zthreads). The latter only has any effect if the first is enabled, which is indeed cfg paralllel_compiler.

Reply:
Ok, it sounds like a) the question only matters when `parallel_compiler` is set, and then separately b) it might be nice to disable thread-based logging when `cfg(parallel_compiler)` is set but `-Z threads` isn't passed. I implemented a) and I think I'll leave b) for when parallel_compiler is better maintained - how does that sound?